components/slider/nz-slider-handle.component.ts

```diff
@@ -1,78 +1,126 @@
-    const show = toBoolean(value);
-    if (this.tooltip) {
-      if (show) {
-        this.tooltip.show();
-      } else {
-        this.tooltip.hide();
-      }
-    }
-  }
+  @Input() nzValue: number;
+  @Input() nzTooltipVisible: SliderShowTooltip = 'default';
+  @Input() nzTipFormatter: (value: number) => string;
+  @Input() @InputBoolean() nzActive = false;
 
-  // Locals
-  @ViewChild('tooltip') tooltip: NzToolTipComponent; // [For tooltip]
-  tooltipTitle: string; // [For tooltip]
+  tooltipTitle: string;
   style: object = {};
 
-  constructor(private _slider: NzSliderComponent) {
+  private hovers_ = new Subscription();
+
+  constructor(
+    private sliderComponent: NzSliderComponent,
+    private ngZone: NgZone,
+    private el: ElementRef,
+    private cdr: ChangeDetectorRef
+  ) {
   }
 
   ngOnChanges(changes: SimpleChanges): void {
-    if (changes.nzOffset) {
-      this._updateStyle();
+    const { nzOffset, nzValue, nzActive, nzTooltipVisible } = changes;
+
+    if (nzOffset) {
+      this.updateStyle();
+    }
+    if (nzValue) {
+      this.updateTooltipTitle();
+      this.updateTooltipPosition();
     }
-    if (changes.nzValue) {
-      this._updateTooltipTitle(); // [For tooltip]
-      this._updateTooltipPosition(); // [For tooltip]
+    if (nzActive) {
+      if (nzActive.currentValue) {
+        this.toggleTooltip(true);
+      } else {
+        this.toggleTooltip(false);
+      }
+    }
+    if (nzTooltipVisible && !nzTooltipVisible.isFirstChange()) {
+      this.tooltip.show();
     }
   }
 
-  // Hover to toggle tooltip when not dragging
-  @HostListener('mouseenter', [ '$event' ])
-  onMouseEnter($event: MouseEvent): void {
-    if (!this._slider.isDragging) {
-      this.nzActive = true;
+  ngAfterViewInit(): void {
+    if (this.nzTooltipVisible === 'always' && this.tooltip) {
+      Promise.resolve().then(() => this.tooltip.show());
     }
+
+    // NOTE: run outside of Angular for performance consideration.
+    this.ngZone.runOutsideAngular(() => {
+      this.hovers_.add(fromEvent(this.el.nativeElement, 'mouseenter').subscribe(() => {
+        if (!this.sliderComponent.isDragging) {
+          this.toggleTooltip(true);
+          this.updateTooltipPosition();
+          this.cdr.detectChanges();
+        }
+      }));
+      this.hovers_.add(fromEvent(this.el.nativeElement, 'mouseleave').subscribe(() => {
+        if (!this.sliderComponent.isDragging) {
+          this.toggleTooltip(false);
+          this.cdr.detectChanges();
+        }
+      }));
+    });
   }
 
-  @HostListener('mouseleave', [ '$event' ])
-  onMouseLeave($event: MouseEvent): void {
-    if (!this._slider.isDragging) {
-      this.nzActive = false;
+  ngOnDestroy(): void {
+    this.hovers_.unsubscribe();
+  }
+
+  private toggleTooltip(show: boolean): void {
+    if (this.nzTooltipVisible !== 'default' || !this.tooltip) {
+      return;
+    }
+
+    if (show) {
+      this.tooltip.show();
```

Comment:
this.tooltip maybe null

Reply:
```ts
private toggleTooltip(show: boolean): void {
    if (this.nzTooltipVisible !== 'default' || !this.tooltip) {
      return;
    }

    if (show) {
      this.tooltip.show();
    } else {
      this.tooltip.hide();
    }
  }
```

If tooltip is null the method returns ahead of this.